tasks/icinga2-bootstrap.yml

```diff
@@ -0,0 +1,125 @@
+---
+# When a --limit is in place only the facts of
+# the selected hosts are retrieved and we need
+# the facts of the other nodes as they are
+# used as defaults of variables used in the
+# templates
+- name: Gather Facts of the other Icinga nodes
+  setup:
+  run_once: True
+  delegate_facts: True
+  delegate_to: "{{ item }}"
+  loop: >-
+    {{ groups[i2_master_group]|default([])
+    + groups[i2_satellite_group]|default([])
+    + groups[i2_agent_group]|default([]) }}
+  when: hostvars[item].ansible_fqdn is not defined
+
+ # Copying role-vars to host-vars
+ # We do this in a loop over all the nodes as we
+ # need the variables of other nodes in the templates
+ # which we don't get if a --limit is in place
+- name: Set default hostvars for Icinga masters
+  set_fact:
+    i2_master: >-
+      {{ hostvars[item].i2_master|default(True) }}
+    i2_satellite: >-
+      {{ hostvars[item].i2_satellite|default(False) }}
+  run_once: True
+  delegate_facts: True
+  delegate_to: "{{ item }}"
+  loop: "{{ groups[i2_master_group]|default([]) }}"
+
+- name: Set default hostvars for Icinga satellites
+  set_fact:
+    i2_master: >-
+      {{ hostvars[item].i2_master|default(False) }}
+    i2_satellite: >-
+      {{ hostvars[item].i2_satellite|default(True) }}
+    i2_parent_zone: >-
+      {{ hostvars[item].i2_parent_zone|default('master') }}
+  run_once: True
+  delegate_facts: True
+  delegate_to: "{{ item }}"
+  loop: "{{ groups[i2_satellite_group]|default([]) }}"
+
+- name: Set default hostvars for all Icinga nodes
+  set_fact:
+    i2_address: >-
+      {{ hostvars[item].i2_address|default(hostvars[item].ansible_facts.default_ipv4.address) }}
+    i2_api_port: >-
+      {{ hostvars[item].i2_api_port|default(i2_api_port) }}
+    i2_api_host: >-
+      {{ hostvars[item].i2_api_host|default(hostvars[item].ansible_facts.default_ipv4.address) }}
+    i2_hostname: >-
+      {{ hostvars[item].i2_hostname|default(hostvars[item].ansible_fqdn) }}
+    i2_zonename: >-
+      {{ hostvars[item].i2_zonename|default(hostvars[item].ansible_fqdn) }}
+    i2_templates: >-
+      {{ hostvars[item].i2_custom_templates|default({}) }}
+    i2_objects: >-
+      {{ hostvars[item].i2_custom_objects|default({}) }}
+    i2_apply_rules: >-
+      {{ hostvars[item].i2_custom_apply_rules|default({}) }}
+  run_once: True
+  delegate_facts: True
+  delegate_to: "{{ item }}"
+  loop: >-
+    {{ groups[i2_master_group]|default([])
+    + groups[i2_satellite_group]|default([])
+    + groups[i2_agent_group]|default([]) }}
+
+- name: Determine child nodes
+  set_fact:
+    i2_peer_nodes: >-
```

Comment:
Okay, @arjanoosting we also need to discuss this point.
Your change makes this work differently than what I had in mind.
A `peer node` as I see it, is a node on the same level. I'm using it to generate the `zones.conf` files on masters.
So in a set-up with two masters (centos1 and centos2), centos1's `i2_peer_nodes` should have only `centos2` and vice versa.

If the `i2_peer_nodes` contains all nodes in a same zone, then this is the result, with the respective host being present twice, once as `NodeName` and once as `centosX`:

```
[root@centos1 icinga2]# cat zones.conf
// BEGIN Ansible managed block for centos1 (localhost)"
object Endpoint NodeName {
}

object Zone ZoneName {
  endpoints = [ NodeName,
  "centos1", "centos2"]
  }
// END Ansible managed block for centos1 (localhost)"
// BEGIN Ansible managed block for peers or parents
object Endpoint "centos1" {
  host = "10.24.1.181"
  port = "5665"
}
object Endpoint "centos2" {
  host = "10.24.1.189"
  port = "5665"
}
```
How is this (and the task below) working in your set-up? I don't have a case where setting these would be necessary.

Reply:
> A few more things I noticed why running tests on our test-infra today.

This indicates quite clearly that we should really add some tests to make our assumptions known… I've been procrastinating on this way too long.